### ISSUE TITLE
cephmetrics dependencies

### ISSUE THREAD

**Issue by pcuzner** [ceph/cephmetrics#65]
Why does the cephmetrics rpm (delivering the dashboards) have a dependency on the collectors rpm?

Error: Package: cephmetrics-0.1-231_ga895841.el7.centos.x86_64 (/cephmetrics-0.1-231_ga895841.el7.centos.x86_64)
           Requires: cephmetrics-collectors = 0.1-231_ga895841.el7.centos

**Comment by zmc** [comment]
From what I can tell, `cephmetrics-ansible` is the only one you need to initiate deployment. It contains everything in the `ansible` subdirectory.

I'm not sure what the purpose of the base `cephmetrics` package is. Its file list:
```
rpm[zack@magna079 ~]$ rpm -ql cephmetrics
/usr/libexec/cephmetrics/dashUpdater.py
/usr/share/cephmetrics
/usr/share/cephmetrics/dashboards
/usr/share/cephmetrics/dashboards/Dashboard Relationships.jpg
/usr/share/cephmetrics/dashboards/ceph-at-a-glance.json
/usr/share/cephmetrics/dashboards/ceph-backend-storage.json
/usr/share/cephmetrics/dashboards/ceph-frontend.json
/usr/share/cephmetrics/dashboards/ceph-rados.json
/usr/share/cephmetrics/dashboards/ceph-rgw-workload.json
/usr/share/cephmetrics/dashboards/disk-busy-by-server.json
/usr/share/cephmetrics/dashboards/iops-by-server.json
/usr/share/cephmetrics/dashboards/latency-by-server.json
/usr/share/cephmetrics/dashboards/network-usage-by-node.json
/usr/share/cephmetrics/dashboards/osd-node-detail.json
/usr/share/doc/cephmetrics-0.1
/usr/share/doc/cephmetrics-0.1/LICENSE
/usr/share/doc/cephmetrics-0.1/README
/usr/share/doc/cephmetrics-0.1/dashboard.yml
/usr/share/doc/cephmetrics-0.1/grafana
/usr/share/doc/cephmetrics-0.1/grafana/grafana.ini
```

**Comment by b-ranto** [comment]
It is a common packaging practice to tie the (sub-)package versions together to prevent mixing of various versions of the collectors and the dashboard (it is generally easier to maintain/test it that way). In this case, it is not that helpful since you still have several nodes. I can strip it away if you want me to.

**Comment by zmc** [comment]
I do think there's value in having a base package that at least contains the docs. Would it be sane to have the subpackages each depend on the base package - but have the base package contain docs?

**Comment by b-ranto** [comment]
Yeah, that would be doable and from the packaging point probably even preferred since the base package would contain the licence file which iirc should get installed always.

**Comment by b-ranto** [comment]
I think we can just drop the dependency as in

https://github.com/ceph/cephmetrics/pull/89

In this case, it will also help us separate the dependencies for the dashboard and the collectors.

**Comment by pcuzner** [comment]
resolved in #89
